src/components/memorygame/MemoryGameCards/index.jsx

```diff
@@ -0,0 +1,171 @@
+import cardInitialStateGenerator from "../cardInitialStateGenerator";
+
+const NUM_REQUIRED_MATCHES = 2;
+// game example: https://www.brightfocus.org/alzheimers/memory-games/match-pictures
+// instead of squares this game uses cards.
+// onClick card flips to show what is behind the card.
+// matchedCards is an array states that holds successfully matched card states
+// setCompleted is boolean
+// setCompleted is true when all possible matches made
+const MemoryGameCards = ({ setMatchedCards, matchedCards, setCompleted }) => {
+  const [t] = useTranslation();
+  // cardsDataArr example [{description:"...",link:"...",phrase:".."},...]
+  const cardsDataArr = [
+    ...t("memoryGame.cards", {
+      returnObjects: true,
+    }),
+  ];
+  // cardStates is an array of states with multiple properties
+  // state example :
+  // {cardKey:"",img:{src:"",imgKey:""},
+  // isFlipped:"",isMatched:"",description:"",link:"",phrase:""}
+  const [cardStates, setCardStates] = useState(
+    cardInitialStateGenerator(cardsDataArr, NUM_REQUIRED_MATCHES)
+  );
+  // selectedCards is an array of clicked card states(objects).
+  // selectedCards array hold only "NUM_REQUIRED_MATCHES" amount of card states
+  const [selectedCards, setSelectedCards] = useState([]);
+
+  // handle click flips card and adds that card's state to selectedCards
+  const handleClick = (clickedCardState) => {
+    const copyOfCardStates = [...cardStates];
+    // current index is needed to keep the order of cards displayed
+    // because goal is to make user remember the spots of previous cards
+    const currentIndex = cardStates.findIndex(
+      (state) => state.cardKey === clickedCardState.cardKey
+    );
+    copyOfCardStates[currentIndex].isFlipped = true;
+    setCardStates(() => [...copyOfCardStates]);
+    setSelectedCards((prevState) => [...prevState, clickedCardState]);
+  };
+
+  // matched cards should have different cardKey, same image key, isFlipped:true
+  // function below tests it
+  const doesMatchLastCard = (prevCardState, lastCardState) => {
+    return (
+      prevCardState.cardKey !== lastCardState.cardKey &&
+      prevCardState.img.imgKey === lastCardState.img.imgKey &&
+      prevCardState.isFlipped &&
+      lastCardState.isFlipped
+    );
+  };
+
+  // Triggered when user clicks on a card(selectedCards change)
+  // checks selected cards to see if they are a match
+  useEffect(() => {
+    // only triggered if "NUM_REQUIRED_MATCHES" amount of cards are clicked
+    if (selectedCards.length !== NUM_REQUIRED_MATCHES) {
+      return;
+    }
+    // match condition of selectedCards will be based on card that is added last to selectedCards
+    const lastCardState = selectedCards[selectedCards.length - 1];
+    const matchesOfLastCardState = selectedCards.filter((prevCard) =>
+      doesMatchLastCard(prevCard, lastCardState)
+    );
+    // matchesOfLastCardState doesnt include last card
+    const matchedCardsArr = [...matchesOfLastCardState, lastCardState];
```

Comment:
you don't need this variable at all, when matchedCardsArr.length == NUM_REQUIRED_MATCHES its just selected cards. the selected cards and the matched cards are the same

you should be able to do everything with selectedCards

Reply:
`selectedCards `holds clicked cards. `matchedCardsArr` holds only successfuly matched cards. The condition you mentioned checks if match is made. For example imagine that you need 3 matches per image(`const NUM_REQUIRED_MATCHES=3`) and game board is like ` a,a,a,b,b,b,c,c,c`. if you click on `c,a,a` in this order. `selectedCards` will hold `c,a,a` and `matchedCardsArr `will hold `a,a`. So if condition is `(selectedCards.length === NUM_REQUIRED_MATCHES)` , a wrong set of cards will have `isMatched:true`. Also, I can't use `selectedCards.length == NUM_REQUIRED_MATCHES` because there is `if (selectedCards.length !== NUM_REQUIRED_MATCHES) {
            return; }` above. Correct way of checking `if (selectedCards.length === NUM_REQUIRED_MATCHES)` would be `else`. But inside `else` i still have to check `matchedCardsArr.length ` .

Reply:
I should clarify things better in comments. I'm having extremely difficult time making things clearer.

Reply:
PTAL